pkg/jira/issue.go

```diff
@@ -419,3 +419,37 @@ func (c *Client) RemoteLinkIssue(issueID, title, url string) error {
 	}
 	return nil
 }
+
+// WatchIssue adds user as a watcher using PUT /issue/{key}/watchers endpoint.
+func (c *Client) WatchIssue(key, watcher string) error {
+	path := fmt.Sprintf("/issue/%s/watchers", key)
+
+	var (
+		res  *http.Response
+		err  error
+		body []byte
+	)
+
+	body, err = json.Marshal(watcher)
+	if err != nil {
+		return err
+	}
+
+	res, err = c.Post(context.Background(), path, body, Header{
```

Comment:
We need to use v2 version of the API to make sure this works in older Jira versions.

```suggestion
	res, err = c.PostV2(context.Background(), path, body, Header{
```